pywbemtools/pywbemcli/pywbemcli.py

```diff
@@ -559,9 +565,15 @@ def create_server_instance(connection_name):
                 pywbem_server = connections_repo[connection_name]
             except KeyError:
                 raise click.ClickException(
-                    "Connection definition '{}' not found in "
-                    "connections file '{}'".
+                    "Connection name '{}' not found in connections file '{}'".
                     format(connection_name, connections_repo.connections_file))
+            except Exception as exc:
```

Comment:
Expecting exceptions other than KeyError from what appears to be a dictionary access violates the principle of least surprise.

I think the issue is to model the connection repo with dictionary access. At first, it appears to be a nice idea, but with code required on the callers side like this, it creates surprising issues. How about remodeling the access of items in the connections repo with methods?

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
See other comments on exxceptions